src/main/java/mate/academy/hibernate/relations/Main.java

```
@@ -1,33 +1,47 @@
 import org.hibernate.SessionFactory;
 
 public class Main {
     public static void main(String[] args) {
-        // use this session factory when you will initialize service instances
-        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        SessionFactory sessionFactory = null;
+        try {
+            sessionFactory = HibernateUtil.getSessionFactory();
+            Country usa = new Country("USA");
+            CountryService countryService
+                    = new CountryServiceImpl(new CountryDaoImpl(sessionFactory));
+            countryService.add(usa);
 
-        Country usa = new Country("USA");
-        CountryService countryService = null; // TODO: initialize this instance
-        countryService.add(usa);
+            Actor vinDiesel = new Actor("Vin Diesel");
+            vinDiesel.setCountry(usa);
+            ActorService actorService = new ActorServiceImpl(new ActorDaoImpl(sessionFactory));
+            actorService.add(vinDiesel);
 
-        Actor vinDiesel = new Actor("Vin Diesel");
-        vinDiesel.setCountry(usa);
-        ActorService actorService = null; // TODO: initialize this instance
-        actorService.add(vinDiesel);
-
-        Movie fastAndFurious = new Movie("Fast and Furious");
-        fastAndFurious.setActors(List.of(vinDiesel));
-        MovieService movieService = null; // TODO: initialize this instance
-        movieService.add(fastAndFurious);
-        System.out.println(movieService.get(fastAndFurious.getId()));
+            Movie fastAndFurious = new Movie("Fast and Furious");
+            fastAndFurious.setActors(List.of(vinDiesel));
+            MovieService movieService = new MovieServiceImpl(new MovieDaoImpl(sessionFactory));
+            movieService.add(fastAndFurious);
+            System.out.println(movieService.get(fastAndFurious.getId()));
+        } catch (Exception e) {
+            throw new RuntimeException("Cannot initialize the sessionFactory");
```

When rethrowing the RuntimeException in the catch block, you discard the original exception cause. Preserve the original exception by using the constructor that accepts a cause, e.g. `throw new RuntimeException("Cannot initialize the sessionFactory", e);`. That keeps the stacktrace and makes debugging much easier. This is important because getting the real cause (configuration, DB connectivity, mapping errors) is necessary to make Main run end-to-end as required. See the catch/rethrow at lines below.